openmoc/map_to_dict.i

```diff
@@ -126,13 +126,65 @@
     }
 
     /* Loop over y */
-    for (int j = 0; j < $2; j++) {
+    for (int i =0; i < $2; i++) {
       /* Extract the value from the list at this location and convert
        * SWIG wrapper to pointer to underlying C++ class instance */
-      PyObject* o = PyList_GetItem(outer_list, j);
+      PyObject* o = PyList_GetItem(outer_list, i);
       void *p1 = 0;
       SWIG_ConvertPtr(o, &p1, SWIGTYPE_p_Universe, 0 | 0);
-      $3[i*$2+j] = (Universe*) p1;
+      $3[j*$2+i] = (Universe*) p1;
+    }
+  }
+}
+
+
+/* Typemap for Lattice::setUniverses3D(int num_z, int num_y, int num_x, 
```

Comment:
Perhaps it would be cleaner to require all lattices to be 3D so that we simply have a single `Lattice::setUniverses(...)` routine? This would require adding an extra level of brackets around the 2D lattice universe lists to make them 3D in all input files. What do you think @samuelshaner @geogunow?